install.go

```diff
@@ -27,147 +38,216 @@ func install(parser *arguments) error {
 	}
 
 	if len(aurs) != 0 {
-		err := aurInstall(aurs, []string{})
+		//todo make pretty
+		fmt.Println("Resolving Dependencies")
+		
+		dt, err := getDepTree(aurs)
 		if err != nil {
-			fmt.Println("Error installing aur packages.")
+			return err
 		}
-	}
-	return nil
-}
 
-// Install sends system commands to make and install a package from pkgName
-func aurInstall(pkgName []string, flags []string) (err error) {
-	q, err := rpc.Info(pkgName)
-	if err != nil {
-		return
-	}
+		if len(dt.Missing) >  0 {
+			fmt.Println(dt.Missing)
+			return fmt.Errorf("Could not find all Deps")
+		}
+		
+		dc, err := getDepCatagories(aurs, dt)
+		if err != nil {
+			return err
+		}
+
+		for _, pkg := range dc.AurMake {
+			if pkg.Maintainer == "" {
+				fmt.Printf("\x1b[1;31;40m==> Warning:\x1b[0;;40m %s is orphaned.\x1b[0m\n", pkg.Name + "-" + pkg.Version)
+			}
+		}
+
+		for _, pkg := range dc.Aur {
+			if pkg.Maintainer == "" {
+				fmt.Printf("\x1b[1;31;40m==> Warning:\x1b[0;;40m %s is orphaned.\x1b[0m\n", pkg.Name + "-" + pkg.Version)
+			}
+		}
+
+		fmt.Println()
 
-	if len(q) != len(pkgName) {
-		fmt.Printf("Some packages from list\n%+v\n do not exist", pkgName)
-	}
 
-	var finalrm []string
-	for _, i := range q {
-		mrm, err := PkgInstall(&i, flags)
+		p1 := func(a []*alpm.Package) {
+			for _, v := range a {
+				fmt.Print("  ", v.Name())
+			}
+		}
+
+		p2 := func(a []*rpc.Pkg) {
+			for _, v := range a {
+				fmt.Print("  ", v.Name)
+			}
+		}
+
+		fmt.Print("Repo (" + strconv.Itoa(len(dc.Repo)) + "):")
+		p1(dc.Repo)
+		fmt.Println()
+
+		fmt.Print("Repo Make (" + strconv.Itoa(len(dc.RepoMake)) + "):")
+		p1(dc.RepoMake)
+		fmt.Println()
+
+		fmt.Print("Aur (" + strconv.Itoa(len(dc.Aur)) + "):")
+		p2(dc.Aur)
+		fmt.Println()
+
+		fmt.Print("Aur Make (" + strconv.Itoa(len(dc.AurMake)) + "):")
+		p2(dc.AurMake)
+		fmt.Println()
+
+		fmt.Println()
+	
+		askCleanBuilds(dc.AurMake)
+		askCleanBuilds(dc.Aur)
+
+		fmt.Println()
+
+		if !continueTask("Proceed with download?", "nN") {
```

Comment:
@Morganamilo this is a new confirmation comparing to the latest release, why did you add it? I was just asked about clean builds, if I wanted to cancel I would have simply pressed <kbd>Ctrl+C</kbd>. I'm happy that all questions are now being asked in the beginning, but I feel this is a question that can be avoided.

/cc @Jguer